### DISCUSSION TITLE
How do I list users then terminate a user's sessions?

### DISCUSSION THREAD

**Discussion by sayerbartlett** [ucfopen/canvasapi#593] [Q&A] [answered]
Hello,

I am trying to figure out how to list users with this endpoint using the python canvasapi:
https://canvas.beta.instructure.com/doc/api/users.html?_gl=1*yqg8fi*_ga*ODA5NDI1NjY3LjE2MzMwMzY4MDA.*_ga_75H5134F9J*MTY3ODM5OTY1MS4xLjEuMTY3ODQwMDUyMC40NC4wLjA.#method.users.api_index

My ultimate goal is to terminate sessions for a user:
https://canvas.beta.instructure.com/doc/api/users.html?_gl=1*yqg8fi*_ga*ODA5NDI1NjY3LjE2MzMwMzY4MDA.*_ga_75H5134F9J*MTY3ODM5OTY1MS4xLjEuMTY3ODQwMDUyMC40NC4wLjA.#method.users.terminate_sessions

I know how to get a user by ID, but I don't know the ID of the user so believe I need to search for them first by using the endpoint mentioned above. Is it possible to get a user directly by passing the user's email? All of the examples I see use user ID but the ID isn't obtained from anywhere, it is just hard coded.

Thanks for your help!

**Comment by bennettscience** [comment, marked as answer]
Yes, you can use that first endpoint to find the user. The `search_term` kwarg can be used and it should be as specific as possible, like a full email address. From the Canvas docs (my emphasis added):

> It will only search against other fields if non-numeric in form, or if the numeric value doesn't yield any matches. **Queries by administrative users will search on SIS ID, Integration ID, login ID, name, or email address.**

So, if you know your user's email address, you can do something like this:

```python
# Set up your canvas instance, etc
account = canvas.get_account(1)
user = account.get_users(search_term="myemail@example.com")

# get_users returns a paginated list, so you can check by iterating
# If more than one user is returned, be sure to check that it's the person you want
for u in user:
    print(u.name)
    # Do stuff
```

**Comment by sayerbartlett** [comment]
Great thanks! I am able to get a canvasapi.user.User object now by searching for email.

Now I can't find how to terminate the user's sessions. There doesn't appear to be anything in the User object. I reviewed the code but maybe I missed it. How can I terminate all sessions for a user?

**Reply by bennettscience** [reply]
Yep, looks like that endpoint is missing. I've created an issue (#594) to log that it needs to be added. In the meantime, you can use your `User._requester` method to build your own request like this:

```python
# get your user
response = user._requester.request(
    "DELETE",
    "/users/{}/sessions".format(self.id)
)
```

Inspecting `response.json()` will give you the result. The Canvas API docs don't say what to expect, though. I would assume `response.status_code` would be a `200`.

**Reply by sayerbartlett** [reply]
Perfect. That works great for now. And you are correct, `response.status_code` is `200`

Thanks for the prompt assistance!
